pyomo/contrib/cspline_external/cspline_parameters.py

```diff
@@ -0,0 +1,316 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2025
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from pyomo.common.dependencies import attempt_import
+from pyomo.common.dependencies import numpy as np
+import pyomo.environ as pyo
+
+
+def _f_cubic(x, alpha, s=None):
+    """
+    Cubic function:
+        y = a1 + a2 * x + a3 * x^2 + a4 * x^3
+
+    Optionally if s is provided it is a segment index.
+        y = a1[s] + a2[s] * x + a3[s] * x^2 + a4[s] * x^3
+
+    This is used to write constraints more compactly.
+
+    Args:
+        x: x variable, numeric, numpy array, or Pyomo component
+        alpha: cubic parameters, numeric or Pyomo component
+        s: optional segment index
+
+    Returns:
+        Pyomo expression, numpy array, or float
+    """
+    if s is None:
+        return alpha[1] + alpha[2] * x + alpha[3] * x**2 + alpha[4] * x**3
+    return alpha[s, 1] + alpha[s, 2] * x + alpha[s, 3] * x**2 + alpha[s, 4] * x**3
+
+
+def _fx_cubic(x, alpha, s=None):
+    """
+    Cubic function first derivative:
+        dy/dx = a2 + 2 * a3 * x + 3 * a4 * x^2
+
+    Optionally if s is provided it is a segment index.
+        dy/dx = a2[s] + 2 * a3[s] * x + 3 * a4[s] * x^2
+
+    This is used to write constraints more compactly.
+
+    Args:
+        x: x variable, numeric, numpy array, or Pyomo component
+        alpha: cubic parameters, numeric or Pyomo component
+        s: optional segment index
+
+    Returns:
+        Pyomo expression, numpy array, or float
+    """
+    if s is None:
+        return alpha[2] + 2 * alpha[3] * x + 3 * alpha[4] * x**2
+    return alpha[s, 2] + 2 * alpha[s, 3] * x + 3 * alpha[s, 4] * x**2
+
+
+def _fxx_cubic(x, alpha, s=None):
+    """
+    Cubic function second derivative:
+        d2y/dx2 = 2 * a3 + 6 * a4 * x
+
+    Optionally if s is provided it is a segment index.
+        d2y/dx2 = 2 * a3[s] + 6 * a4[s] * x
+
+    This is used to write constraints more compactly.
+
+    Args:
+        x: x variable, numeric, numpy array, or Pyomo component
+        alpha: cubic parameters, numeric or Pyomo component
+        s: optional segment index
+
+    Returns:
+        Pyomo expression, numpy array, or float
+    """
+    if s is None:
+        return 2 * alpha[3] + 6 * alpha[4] * x
+    return 2 * alpha[s, 3] + 6 * alpha[s, 4] * x
+
+
+class CsplineParameters(object):
```

Comment:
```suggestion
class CsplineParameters:
```
`(object)` isn't necessary in Python3.